irods/src/main/java/ch/cyberduck/core/irods/IRODSConnectionUtils.java

```diff
@@ -0,0 +1,112 @@
+package ch.cyberduck.core.irods;
+
+/*
+ * Copyright (c) 2002-2025 iterate GmbH. All rights reserved.
+ * https://cyberduck.io/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+import ch.cyberduck.core.preferences.HostPreferencesFactory;
+import ch.cyberduck.core.preferences.PreferencesReader;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.irods.irods4j.authentication.AuthPlugin;
+import org.irods.irods4j.authentication.NativeAuthPlugin;
+import org.irods.irods4j.authentication.PamPasswordAuthPlugin;
+import org.irods.irods4j.high_level.connection.IRODSConnection;
+import org.irods.irods4j.high_level.connection.IRODSConnectionPool;
+import org.irods.irods4j.high_level.connection.QualifiedUsername;
+import org.irods.irods4j.low_level.api.IRODSApi;
+import org.irods.irods4j.low_level.api.IRODSException;
+
+import java.io.IOException;
+
+final class IRODSConnectionUtils {
+
+    private static final Logger log = LogManager.getLogger(IRODSConnectionUtils.class);
+
+    public static IRODSApi.ConnectionOptions initConnectionOptions(IRODSSession session) {
+        log.debug("configuring iRODS connection.");
+        final PreferencesReader preferences = HostPreferencesFactory.get(session.getHost());
+        final IRODSApi.ConnectionOptions options = new IRODSApi.ConnectionOptions();
+
+        options.clientServerNegotiation = preferences.getProperty(IRODSProtocol.CLIENT_SERVER_NEGOTIATION);
+        options.sslProtocol = preferences.getProperty(IRODSProtocol.TLS_PROTOCOL);
+        options.sslTruststore = preferences.getProperty(IRODSProtocol.TLS_TRUSTSTORE);
```

Comment:
We don't want to support a trust store referenced like this but instead it should be possible to inject the trust store configured depending on the runtime environment you can obtain from the `IRODSSession` constructor parameter.

Reply:
Do you have an example demonstrating that?

Reply:
@dkocher How do I test SSL/TLS in Cyberduck?

Does Cyberduck configure the TrustManager for the user?

Is there documentation showing how to enable it?

Reply:
* If you support mTLS, it is required the certificate can be found in the user trust store on the system, e.g. the login keychain on macOS or the Windows Certificate Store.
* To verify certificates the chain must be trusted, e.g. intermediate and/or root certificates are found in the system trust store. Otherwise the user is prompted to proceed when missing trust.

Reply:
Based on what you said, my understanding is that Cyberduck requires certificates to be installed into a standard location.

Hypothetically, what happens if a user isn't allowed to install the necessary certs into the standard location?

> Otherwise the user is prompted to proceed when missing trust.

Can you say more about this? What does it mean to "proceed with missing trust"?